examples/ansible/code/Synchronize_DNS_Zones

```diff
@@ -0,0 +1,109 @@
+- name: Synchronize DNS
+  hosts: 24-bind-01
+  vars:
+    netbox_url: 'https://<netbox-url>/'
+    netbox_token: '<redacted>'
+  tasks:
+
+#set Variables
+    - name: get active Zone Names
+      ansible.builtin.set_fact:
+        zone_names: "{{ query('netbox.netbox.nb_lookup', 'zones', plugin='netbox_dns',
+                           api_endpoint=netbox_url,
+                           api_filter='active=true',
+                           token=netbox_token)
+                    | map(attribute='value') | map(attribute='name') }}"
+
+    - name: get used Views
+      ansible.builtin.set_fact:
+        used_views: "{{ query('netbox.netbox.nb_lookup', 'views', plugin='netbox_dns',
+                           api_endpoint=netbox_url,
+                           token=netbox_token)
+                    | map(attribute='value')| map(attribute='name') }}"
+
+#Create View-Folder and remove unwanted Files
+    - name: Create View-Directory
+      ansible.builtin.file:
+        path: "/path_to_your_zone_files/{{ item }}"
+        state: directory
+        mode: '755'
+      loop: "{{ used_views }}"
+
+    - name: Collect Directory contents
+      ansible.builtin.shell: >
+        ls -1 /path_to_your_zone_files/
+      register: directories
+      changed_when: False
+
+    - name: Remove unwanted Content
+      ansible.builtin.file:
+        path: "/path_to_your_zone_files/{{ item }}"
+        state: absent
+      with_items: "{{ directories.stdout_lines }}"
+      when: >
+        item not in used_views
+
+#Generate Zones for Views
+    - name: Generate Zone Files
+      ansible.builtin.template:
+          src: dns.zone.j2
+          dest: "/path_to_your_zone_files/{{ current_view }}/{{ item }}.zone"
+      vars:
+          zonename: "{{ item }}"
+          zone: "{{ query('netbox.netbox.nb_lookup', 'zones', plugin='netbox_dns',
+                           api_endpoint=netbox_url,
+                           api_filter='name='+zonename,
+                           token=netbox_token)
+                    | map(attribute='value')
+                    | first }}"
+          records: "{{ query('netbox.netbox.nb_lookup', 'records', plugin='netbox_dns',
+                             api_endpoint=netbox_url,
+                             api_filter='view='+current_view+' zone='+zone.name,
+                             token=netbox_token)
+                       | map(attribute='value') }}"
+          current_view: "{{ zone.view.name }}"
+      loop: "{{ zone_names }}"
+      notify: restart bind9
+
+#Delete not needed Zones per View
+    - name: Initialize lists for later use
+      ansible.builtin.set_fact:
+        present_files: []
+        expected_files: []
+        delete_files_list: []
+
+    - name: Collect Directory contents
+      ansible.builtin.shell: >
+        ls -1 /path_to_your_zone_files/{{ item }} | sed 's/\.[^.]*$//'
```

Comment:
This is one of the places where using `ansible.builtin.shell` can be tempting (but you'll admit it's not exactly readable) :-)

`ansible.builtin.find` together with the `basename` and/or `splitext` / `first` filter should do the trick here.